packages/nodes-base/nodes/Transform/RemoveDuplicates/v2/RemoveDuplicatesV2.node.ts

```diff
@@ -0,0 +1,374 @@
+						'Fields To Exclude',
+					);
+
+					if (!fieldsToExclude.length) {
+						throw new NodeOperationError(
+							this.getNode(),
+							'No fields specified. Please add a field to exclude from comparison',
+						);
+					}
+					if (!disableDotNotation) {
+						keys = Object.keys(flattenKeys(items[0].json));
+					}
+					keys = keys.filter((key) => !fieldsToExclude.includes(key));
+				}
+				if (compare === 'selectedFields') {
+					const fieldsToCompare = prepareFieldsArray(
+						this.getNodeParameter('fieldsToCompare', 0, '') as string,
+						'Fields To Compare',
+					);
+					if (!fieldsToCompare.length) {
+						throw new NodeOperationError(
+							this.getNode(),
+							'No fields specified. Please add a field to compare on',
+						);
+					}
+					if (!disableDotNotation) {
+						keys = Object.keys(flattenKeys(items[0].json));
+					}
+					keys = fieldsToCompare.map((key) => key.trim());
+				}
+
+				// This solution is O(nlogn)
+				// add original index to the items
+				const newItems = items.map(
+					(item, index) =>
+						({
+							json: { ...item.json, __INDEX: index },
+							pairedItem: { item: index },
+						}) as INodeExecutionData,
+				);
+				//sort items using the compare keys
+				newItems.sort((a, b) => {
+					let result = 0;
+
+					for (const key of keys) {
+						let equal;
+						if (!disableDotNotation) {
+							equal = isEqual(get(a.json, key), get(b.json, key));
+						} else {
+							equal = isEqual(a.json[key], b.json[key]);
+						}
+						if (!equal) {
+							let lessThan;
+							if (!disableDotNotation) {
+								lessThan = lt(get(a.json, key), get(b.json, key));
+							} else {
+								lessThan = lt(a.json[key], b.json[key]);
+							}
+							result = lessThan ? -1 : 1;
+							break;
+						}
+					}
+					return result;
+				});
+
+				validateInputData(this.getNode(), newItems, keys, disableDotNotation);
+
+				// collect the original indexes of items to be removed
+				const removedIndexes: number[] = [];
+				let temp = newItems[0];
+				for (let index = 1; index < newItems.length; index++) {
+					if (compareItems(newItems[index], temp, keys, disableDotNotation)) {
+						removedIndexes.push(newItems[index].json.__INDEX as unknown as number);
+					} else {
+						temp = newItems[index];
+					}
+				}
+				let updatedItems: INodeExecutionData[] = items.filter(
+					(_, index) => !removedIndexes.includes(index),
+				);
+
+				if (removeOtherFields) {
+					updatedItems = updatedItems.map((item, index) => ({
+						json: pick(item.json, ...keys),
+						pairedItem: { item: index },
+					}));
+				}
+				returnData.push(updatedItems);
+
+				return returnData;
+			}
+			case 'removeItemsSeenInPreviousExecutions': {
+				const logic = this.getNodeParameter('logic', 0);
+				if (logic === 'removeItemsWithAlreadySeenKeyValues') {
+					const context = this.getNodeParameter('options.scope', 0, 'node');
+
+					if (!['node', 'workflow'].includes(context as string)) {
```

Comment:
Nicer to use `getNodeParameter<ProcessedDataContext>` instead of casting every usage

Reply:
getNodeParameter<T> seems to be designed to work with resource parameter